### PR TITLE
Add check for required params

### LINKED ISSUE #98: Add check that all required params are non-null
Would help me avoid some weird errors later.

posting this here so I remember to do it

### PR DESCRIPTION
**Pull request by cookkate**
in progress

fixes #98 

This is mostly a Kate's quality of life change, since I forget to set things a lot. Whether it's useful or we just forget to keep the 'required params' list up to date remains to be seen.

### CODE REVIEW

**Comment by mattjmeier** [comment]
I re-ran the tests and it looks like this one is failing because of the function added (check_required_params) - any idea why that would happen?

**Review by mattjmeier — CHANGES_REQUESTED** [review]
Should figure out why it's crashing before merging I think

File: scripts/run_DESeq2.R [hunk 1]
```diff
@@ -48,6 +48,7 @@ params <- set_up_platform_params(params)
 # Set this variable to be TRUE if you want to have separate plots of top genes as defined in the R-ODAF template
 params$R_ODAF_plots <- FALSE
 
+check_required_params()
```
Comment (mattjmeier) [inline scripts/run_DESeq2.R:51]: I guess when you call the function without params, it doesn't have them in memory when setup_functions.R gets sourced... might be able to fix with source(local = T)... this works for you locally though?

Comment (cookkate) [reply scripts/run_DESeq2.R]: whoops. That's a bug, should be passing in the params.

**Pull request merged**